Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -8999,5 +9071,30 @@ private void VerticalScrollBar_Scroll(object sender, ScrollEventArgs e)
         {
             ProcessVerticalScroll(e.ScrollEventType);
         }
+
+        private bool _loadingItems;
+
+        private async void LoadMoreDataFromIncrementalItemsSourceAsync()
+        {
+            if (IncrementalLoadingTrigger == IncrementalLoadingTrigger.Edge && DataConnection.IsDataSourceIncremental && DataConnection.HasMoreItems && !_loadingItems)
+            {
+                var bottomScrollOffHeight = Math.Max(0, EdgedRowsHeightCalculated - CellsHeight - VerticalOffset);
+
+                if ((IncrementalLoadingThreshold * CellsHeight) >= bottomScrollOffHeight)
+                {
+                    _loadingItems = true;
+
+                    var numberOfRowsToLoad = (int)(DataFetchSize * CellsHeight / RowHeightEstimate);
+
+                    DataConnection.LoadingOperation = numberOfRowsToLoad > 1
+                        ? DataConnection.LoadMoreItemsAsync((uint)numberOfRowsToLoad)
+                        : DataConnection.LoadMoreItemsAsync(1);
```

Comment:
Should simply be "DataConnection.LoadMoreItems((uint)numberOfRowsToLoad);" as a void method.
No need for _loadingItems flag, as long as we use the _loadingOperation instead (you'll see what I mean in my other comments).